src/core/qgsexpression.cpp

```diff
@@ -2218,7 +2218,7 @@ bool QgsExpression::NodeUnaryOperator::prepare( QgsExpression* parent, const Qgs
 
 QString QgsExpression::NodeUnaryOperator::dump() const
 {
-  return QString( "%1 %2" ).arg( UnaryOperatorText[mOp] ).arg( mOperand->dump() );
+  return QString( "%1%2" ).arg( UnaryOperatorText[mOp] ).arg( mOperand->dump() );
```

Comment:
This will also output `NOT` with no space which isn't correct.  I was thinking about this and I'm not sure this needs to change even though I though it was wrong at the start.